CorsixTH/Lua/entities/machine.lua

```diff
@@ -154,14 +175,15 @@ function Machine:calculateSmoke(room)
   local threshold = self:getRemainingUses()
 
   -- If now exploding, clear any smoke
-  if threshold < 1 then
+  if threshold < 1 and self.saved_by_extinguisher == 0 then
```

Comment:
pass explode into this function, or don't call calculateSmoke when explode is true, just setSmoke directly. It's kinda weird how we set smoke here and everything else in the other function.

Reply:
This is to do with a function in room.lua . If we edit the room, the smoke has to be reset.

Reply:
In terms of the Machine:machineUsed(room) function it is easier to set the smoke inside there, especially with now checking if an extinguisher saved the room. Nonetheless I've removed an offending bit of the if statement that would never be used in calculateSmoke(room)